client/src/utils/filters/filterDoctors.js

```diff
@@ -1,62 +1,87 @@
-function withinFee(feeValue, doctorFee){
-    if(doctorFee <= feeValue[1] && doctorFee >= feeValue[0]){
-        return true;
-    } else {
-        return false;
-    }
-}
+import distanceBetweenGeoPoints from "../functions/GeoDistance.js";
 
-function hasSpecialization(filterSpecialization, doctorSpecialization){
-    if(filterSpecialization.length === 0){
-        return true;
-    }
-    for(let i = 0 ; i < filterSpecialization.length ; i++ ){
-        if(!doctorSpecialization.includes(filterSpecialization[i])){
-            return false;
-        };
-    }
+function withinFee(feeValue, doctorFee) {
+  if (doctorFee <= feeValue[1] && doctorFee >= feeValue[0]) {
     return true;
+  } else {
+    return false;
+  }
 }
 
-function hasTreatment(filterTreatment, doctorTreatment){
-    if(filterTreatment.length === 0){
-        return true;
-    }
-    for(let i = 0 ; i < filterTreatment.length ; i++ ){
-        if(!doctorTreatment.includes(filterTreatment[i])){
-            return false;
-        };
+function hasSpecialization(filterSpecialization, doctorSpecialization) {
+  if (filterSpecialization.length === 0) {
+    return true;
+  }
+  for (let i = 0; i < filterSpecialization.length; i++) {
+    if (!doctorSpecialization.includes(filterSpecialization[i])) {
+      return false;
     }
+  }
+  return true;
+}
+
+function hasTreatment(filterTreatment, doctorTreatment) {
+  if (filterTreatment.length === 0) {
     return true;
+  }
+  for (let i = 0; i < filterTreatment.length; i++) {
+    if (!doctorTreatment.includes(filterTreatment[i])) {
+      return false;
+    }
+  }
+  return true;
 }
 
 // Location filter logic is on hold. Can do with geo location or comparing strings
 
 function filterDoctors(
+  currentUser,
+  currentUserData,
+  searchDoctor,
   doctorList,
   location,
   feeValue,
   specializations,
   treatments,
+  SortBy
 ) {
-    const doctorCount = doctorList.length;
-    const filteredList = [];
-
-    for(let i = 0 ; i < doctorCount; i++) {
-        const filterDoc = doctorList[i];
-        if(!withinFee(feeValue, filterDoc.consultationFee)){
-            continue;
-        }
-        if(!hasSpecialization(specializations,filterDoc.specialization)){
-            continue;
-        }
-        if(!hasTreatment(treatments, filterDoc.treatments)){
-            continue;
-        }
-        filteredList.push(filterDoc);
+
+  const doctorCount = doctorList.length;
+  const filteredList = [];
+
+  for (let i = 0; i < doctorCount; i++) {
+    const filterDoc = doctorList[i];
+    if(!searchDoctor === "" || !filterDoc.doctorName.toLowerCase().includes(searchDoctor.toLowerCase())) {
+        continue;
+    }
+    if (!withinFee(feeValue, filterDoc.consultationFee)) {
+      continue;
+    }
+    if (!hasSpecialization(specializations, filterDoc.specialization)) {
+      continue;
     }
+    if (!hasTreatment(treatments, filterDoc.treatments)) {
+      continue;
+    }
```

Comment:
Can make a single if statement spanning lines. Multiple branching if statements are hard to follow, grouping them by action like rejection here linearizes them a bit. To organize different criteria of rejection newlines can be used but usually not necessary.
Example,
```
if(!searchDoctor === "" || !filterDoc.doctorName.toLowerCase().includes(searchDoctor.toLowerCase())
    || !withinFee(feeValue, filterDoc.consultationFee) 
    || !hasSpecialization(specializations, filterDoc.specialization)
    || !hasTreatment(treatments, filterDoc.treatments)) {
      continue;
    }
```

Reply:
I see...